nodedb.go

```diff
@@ -363,14 +394,94 @@ func (ndb *nodeDB) deleteVersion(version int64) error {
 	if err != nil {
 		return err
 	}
-	if rootKey == nil || rootKey.version < version {
-		if err := ndb.batch.Delete(ndb.nodeKey(&NodeKey{version: version, nonce: 1})); err != nil {
+	if rootKey == nil || GetNodeKey(rootKey).version < version {
+		if err := ndb.batch.Delete(ndb.nodeKey(GetRootKey(version))); err != nil {
 			return err
 		}
 	}
 
-	return ndb.traverseOrphans(version, func(orphan *Node) error {
-		return ndb.batch.Delete(ndb.nodeKey(orphan.nodeKey))
+	return ndb.traverseOrphans(version, version+1, func(orphan *Node) error {
+		return ndb.batch.Delete(ndb.nodeKey(orphan.GetKey()))
+	})
+}
+
+// deleteLegacyNodes deletes all legacy nodes with the given version from disk.
+// NOTE: This is only used for DeleteVersionsFrom.
+func (ndb *nodeDB) deleteLegacyNodes(version int64, nk []byte) error {
```

Comment:
Is there test coverage for these new `deleteLegacy*` methods?

Reply:
yeah, it will be ussed in `DeleteVersionsTo` and it is covered in `migrate_test.go`